site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -137,6 +137,17 @@ private function loadSubmissionFiles() {
                     $this->files[$dir][0][$file] = $details;
                 }
             }
+
+            if ($dir === 'submissions') {
+                $user_assignment_settings_path = FileUtils::joinPaths($course_path, $dir, $gradeable->getId(), $submitter_id);
+                $user_assignment_settings = FileUtils::getAllFiles($user_assignment_settings_path, [], true);
+                foreach ($user_assignment_settings as $file => $details) {
+                    if (basename($file) === 'user_assignment_settings.json') {
+                        $this->meta_files[$dir][$file] = $details;
+                    }
+                }
```

Comment:
Rather than looping over all files in that directory, you could check the files existence right away.